darshan-util/pydarshan/darshan/experimental/plots/plot_dxt_heatmap.py

```diff
@@ -320,16 +320,18 @@ def plot_heatmap(
     Raises
     ------
 
-    NotImplementedError: raised if "DXT_POSIX" is not in the input modules.
+    NotImplementedError: if a DXT module is not input (i.e. "DXT_POSIX") or if
+    the module is not in the input ``DarshanReport``.
 
     """
-    if "DXT_POSIX" not in mods:
-        # TODO: for the moment reject any cases that don't input "DXT_POSIX"
-        # until we can properly aggregate the data
-        raise NotImplementedError("DXT_POSIX module is required.")
+    if "DXT" not in mod:
+        raise NotImplementedError(f"Only DXT modules are supported.")
+
+    if mod not in report.modules:
+        raise NotImplementedError(f"Module {mod} not found in DarshanReport.")
```

Comment:
Hmm, is something really not implemented here? It sound like maybe some data or values are not available in the report, so an error type reflecting that situation might be more appropriate?

Reply:
I agree that `NotImplementedError` isn't really appropriate. I think for a case like this a custom exception, like a Darshan-specific `ModuleNotFoundError`, would be nice to have. I'll change this to a `ValueError` for now I think.

Reply:
I changed this to a `ValueError` for now, but let me know if you think there is a better alternative.